OrbitCaptureClient/CaptureClient.cpp

```diff
@@ -197,34 +223,50 @@ bool CaptureClient::StopCapture() {
     LOG("Finished writing on Capture's gRPC stream: asking to stop capturing");
   }
 
-  state_ = State::kStopping;
   return true;
 }
 
-bool CaptureClient::TryAbortCapture() {
-  absl::MutexLock lock(&state_mutex_);
-  if (state_ != State::kStarted && state_ != State::kStopping) {
-    LOG("TryAbortCapture ignored, because the capture is not started nor stopping");
-    return false;
+bool CaptureClient::AbortCaptureAndWait(int64_t max_wait_ms) {
+  bool try_cancel_called = false;
+  {
+    absl::ReaderMutexLock lock{&context_and_stream_mutex_};
+    if (client_context_ != nullptr) {
+      LOG("Calling TryCancel on Capture's gRPC context: aborting the capture");
+      try_abort_ = true;
+      client_context_->TryCancel();  // reader_writer_->Read in Capture should then fail
+      try_cancel_called = true;
+    }
   }
 
-  CHECK(client_context_ != nullptr);
+  if (!try_cancel_called) {
+    LOG("AbortCaptureAndWait ignored: no ClientContext to TryCancel");
+    return false;
+  }
```

Comment:
maybe move it up and do early return on L233? is there a reason for avoiding read lock while LOGging?

Reply:
(and you can drop try_cancel_called flag)

Reply:
Yeah, in general I try to print logs when not holding a mutex when possible (to do as little as possible when holding the mutex), but I guess it doesn't really matter, and the code improves quite a bit from changing this, so done.